cpp/src/Slice/Parser.cpp

```diff
@@ -1319,22 +1285,23 @@ Slice::Container::createClassDecl(const string& name, bool intf)
     // have a declaration for the class in this container, we don't
     // create another one.
     //
-    for(ContainedList::const_iterator q = _contents.begin(); q != _contents.end(); ++q)
+    for (const auto& q : _contents)
     {
-        if((*q)->name() == name)
+        if(q->name() == name)
         {
-            ClassDeclPtr decl = ClassDeclPtr::dynamicCast(*q);
+            ClassDeclPtr decl = ClassDeclPtr::dynamicCast(q);
             if(decl)
             {
                 return decl;
             }
 
-            assert(ClassDefPtr::dynamicCast(*q));
+            def = ClassDefPtr::dynamicCast(q);
+            assert(def);
```

Comment:
... to here?

The previous logic seemed more correct because `_unit->findContents` will search everything in the unit for a corresponding `ClassDef`,

but here, we iterate through `_contents`, which only searches the contents of the current module.

For example:
```
module Foo { class C; }
module Foo { class C {} }
```
With this change, it doesn't seem like these will be linked together.

Reply:
This change was "imported" from #864.

I think it is correct. Please write a test-case showing it's not.

Reply:
Note that we are in createClassDecl, not createClassDef.